catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/converter/GmdConverter.java

```diff
@@ -256,8 +363,58 @@ protected void addExtent(MetacardImpl metacard, XstreamPathValueTracker pathValu
                 pathValueTracker.add(new Path(GmdConstants.BBOX_EAST_LON_PATH), eastLon);
                 pathValueTracker.add(new Path(GmdConstants.BBOX_SOUTH_LAT_PATH), southLat);
                 pathValueTracker.add(new Path(GmdConstants.BBOX_NORTH_LAT_PATH), northLat);
+            }
+        }
+    }
 
+    private void addStringAttributeToXml(Metacard metacard,
+            XstreamPathValueTracker pathValueTracker, String path, String metacardAttributeName,
+            String fallbackMetacardAttribute) {
+        Attribute attribute = metacard.getAttribute(metacardAttributeName);
+        if (attribute != null) {
+            addStringAttributeToPath(attribute, pathValueTracker, path);
+        } else if (StringUtils.isNotEmpty(fallbackMetacardAttribute)) {
+            Attribute fallbackAttribute = metacard.getAttribute(fallbackMetacardAttribute);
+            if (fallbackAttribute != null) {
+                addStringAttributeToPath(fallbackAttribute, pathValueTracker, path);
             }
         }
     }
+
+    private void addStringAttributeToPath(Attribute attribute,
+            XstreamPathValueTracker pathValueTracker, String path) {
+        Serializable serializable = attribute.getValue();
+        if (serializable instanceof String) {
+            String value = (String) serializable;
+            pathValueTracker.add(new Path(path), value);
+        }
+    }
+
+    private void addDateAttributeToXml(Metacard metacard, XstreamPathValueTracker pathValueTracker,
+            String path, String metcardAttributeName) {
+        GregorianCalendar modifiedCal = new GregorianCalendar();
+        Attribute attribute = metacard.getAttribute(metcardAttributeName);
+        if (attribute != null) {
+            Serializable serializable = attribute.getValue();
+            if (serializable instanceof Date) {
```

Comment:
Could move `modifiedCal` down here since you won't use it unless these conditions are met.